confmap/provider/envprovider/provider_test.go

```diff
@@ -154,6 +154,68 @@ func TestEmptyEnvWithLoggerWarn(t *testing.T) {
 	assert.Equal(t, envName, logLine.Context[0].String)
 }
 
+func TestEnvWithDefaultValue(t *testing.T) {
+	env := createProvider()
+	tests := []struct {
+		name     string
+		unset    bool
+		varValue string
+		uri      string
+		expected string
+	}{
+		{name: "unset", unset: true, uri: "env:MY_VAR:-default % value", expected: "default % value"},
+		{name: "unset2", unset: true, uri: "env:MY_VAR:-", expected: ""}, // empty default still applies
+		{name: "empty", varValue: "", uri: "env:MY_VAR:-foo", expected: ""},
+		{name: "not empty", varValue: "value", uri: "env:MY_VAR:-", expected: "value"},
```

Comment:
Can you add some test cases for:

- `"env:MY_VAR:-test:-test`
- `"env:MY_VAR:?test:?test`
- `"env:-MY_VAR`
- `"env:?MY_VAR`